src/main/java/org/gridsuite/directory/server/DirectoryService.java

```diff
@@ -195,43 +216,79 @@ private void emitDirectoryChanged(UUID directoryUuid, String userId, String erro
             .then();
     }
 
+    //
+    private boolean isElementUpdatable(ElementAttributes element, String userId, boolean forDeletion) {
+        if (element.getType().equals(DIRECTORY)) {
+            return element.getOwner().equals(userId) && // Only owner
+                (!forDeletion || getDirectoryElementsStream(element.getElementUuid())
+                    .filter(e -> e.getType().equals(DIRECTORY))
+                    .allMatch(e -> isElementUpdatable(e, userId, true))
+                );
+        } else {
+            return getParentElement(element.getElementUuid()).isAllowed(userId);
+        }
+    }
```

Comment:
We agreed that if it's public everyone can do whatever they want with it including deleting it